User_Service/Interfaces/Repositories/database.go

```diff
@@ -0,0 +1,99 @@
+package database
+
+import (
+	"log"
+
+	entity "github.com/E-Furqan/Food-Delivery-System/Entity"
+	"gorm.io/gorm"
+)
+
+// Repository struct to handle dependency injection
+type Repository struct {
+	DB *gorm.DB
+}
+
+// NewRepository is a constructor function to initialize the repository with a DB connection
+func NewRepository(db *gorm.DB) *Repository {
+	return &Repository{
+		DB: db,
+	}
+}
+
+// CreateRole inserts a new role into the database
+func (repo *Repository) CreateRole(role *entity.Role) error {
+	result := repo.DB.Create(role)
+	if result.Error != nil {
+		log.Printf("Error creating role: %v", result.Error)
+		return result.Error
+	}
+	return nil
+}
+
+// CreateUser inserts a new user into the database
+func (repo *Repository) CreateUser(user *entity.User) error {
+	result := repo.DB.Create(user)
+	return result.Error
+}
+
+// LoadUserWithRole loads a user with its associated role from the database
+func (repo *Repository) LoadUserWithRole(userID uint) (entity.User, error) {
+	var userWithRole entity.User
+
+	// Load the user with the associated role
+	err := repo.DB.Preload("Role").First(&userWithRole, userID).Error
+	if err != nil {
+		log.Printf("Error loading user with role: %v  %s", err, userWithRole.Role.Role_type)
+		return entity.User{}, err // Return an empty User struct and the error
+	}
+	log.Printf("Error loading user with role: %v  %s", err, userWithRole.Role.Role_type)
+	return userWithRole, err // Return the loaded user and a nil error
+}
+
+// WhereUsername retrieves a user by username
+func (repo *Repository) WhereUsername(username string, user *entity.User) error {
```

Comment:
do you think WhereUsername is a valid name? can't we have a generic method name?